shm/Player.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Player.hpp"
+#include "Ship.hpp"
+#include "Cargo.hpp"
+#include <memory>
+#include <utility>
+#include <vector>
+#include <iostream>
+
+void Player::calculateAvailableSpace() {
+    const auto& cargoVec = ship_->getCargoVec();
+    if(ship_->getCapacity() <= cargoVec.size()) {
+        availableSpace_ = 0;
+    }
+    availableSpace_ = ship_->getCapacity() - cargoVec.size();
+}
+
+Player::Player(std::unique_ptr<Ship> ship, const size_t& money)
+    : ship_{std::move(ship)}, money_{money} {
+        calculateAvailableSpace();
+    }
+
+const std::unique_ptr<Ship>& Player::getShip() const {
```

Comment:
Are you sure this is the best option? Maybe raw ptr is better here.

Reply:
I'm not sure. Could you provide more detailes why raw ptr would be better here? Is it always better to return raw ptr than reference to unique_ptr in similar case? Or just in this case knowing future use of that method? Thanks in advance.

Reply:
Fixed.